staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -413,7 +413,7 @@ func (d *Helper) deletePods(pods []corev1.Pod, getPodFn func(namespace, name str
 
 func waitForDelete(params waitForDeleteParams) ([]corev1.Pod, error) {
 	pods := params.pods
-	err := wait.PollImmediate(params.interval, params.timeout, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(params.ctx, params.interval, params.timeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
Isn't this function used by other commands and how are we sure that `params.ctx` is not nil?

Reply:
Yes, this function is used by other commands, and I would like to migrate them all to use the PollUntilContextTimeout function (pending your approval). The params.ctx is assigned here: [kubernetes/kubernetes] https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L396).
I can also add an additional check (but I don't think its needed).

Reply:
I was asking about `waitForDelete`. I see that 2 places use this function and both of them are passing non-nil context. But in my opinion, not panicking is the responsibility of this function (what happen if somewhere else start using this by passing nil context). So that I'd recommend adding nil check for params.ctx.

Reply:
I understand. Ok I'll adjust.